themes/apiarybasestyles/source/css/buttons.styl

```diff
@@ -10,3 +10,10 @@
     background-color #264653
     &:hover
       background-color #2A9D8F
+
+.loadingIndicator
+  $loadingIndicator()
+  &--big
+    $loadingIndicator(size: 12px)
+  &--blue
+    $loadingIndicator(color: $primaryColor)
```

Comment:
You can't parametrize extends - which is something you want to do when you using base-styles for your UI. Idea is that you can always use base-styles to create basic elements on page from mixins and extend upon those if needed. Plus gzip should take care of any repetition to make overhead tiny. You should give it a try